pkg/apps/util/util.go

```diff
@@ -276,6 +338,72 @@ func CopyApiEnvVarToV1EnvVar(in []api.EnvVar) []v1.EnvVar {
 	return out
 }
 
+func CopyPodTemplateSpecToV1PodTemplateSpec(spec *api.PodTemplateSpec) *v1.PodTemplateSpec {
+	copied, err := api.Scheme.DeepCopy(spec)
+	if err != nil {
+		panic(err)
+	}
+	in := copied.(*api.PodTemplateSpec)
+	out := &v1.PodTemplateSpec{}
+	if err := v1.Convert_api_PodTemplateSpec_To_v1_PodTemplateSpec(in, out, nil); err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// HasLatestPodTemplate checks if the provided replication controller has the latest
+// deployment config pod template. If not it will return false and the string diff.
+func HasLatestPodTemplate(dc *deployapi.DeploymentConfig, rc *v1.ReplicationController) (bool, string, error) {
+	configTemplate := CopyPodTemplateSpecToV1PodTemplateSpec(dc.Spec.Template)
+	configTemplate = filterTemplateAnnotationsAndLabels(configTemplate)
+	rcCopy, err := DeploymentDeepCopyV1(rc)
+	if err != nil {
+		return true, "", err
```

Comment:
deepcopy failing seems more like a panic to me

Reply:
@sttts made changes so that deepcopies should not error out - don't we have that code already in Origin?

Reply:
it will come in with the 3.8 rebase (or @deads2k backporting it, am not through my mails yet after a week vacataion. Maybe we have it already in origin :-) )

Reply:
will rather wait for that than introducing panic here now ;)

Reply:
(also this is pre-existing, other deep copies do the same)